### PR TITLE
feat: support version compatibility check

### PR DESCRIPTION
**Pull request by wingyplus**
Calls `Dagger.Client.check_version_compatibility/2` against the engine
version and warning it if the result is failure. This requires
`Dagger.EngineConn` to expose engine version to make usable in
`Dagger.connect/1`.

### CODE REVIEW

File: sdk/rust/crates/dagger-sdk/src/core/mod.rs
Comment (TomChv) [inline sdk/rust/crates/dagger-sdk/src/core/mod.rs:1]: This change is relative to rust, I think you should open a different PR to include the version compatibility check in another language so we can merge PR fasters :)

**Review by TomChv — COMMENTED** [review]
Changes looks good on Elixir side, I'm ready to approve once you turn the PR ready to merge.

You can open a different PR if you wanna apply changes on the Rust SDK :)

File: sdk/rust/crates/dagger-sdk/src/core/mod.rs
Comment (wingyplus) [reply sdk/rust/crates/dagger-sdk/src/core/mod.rs:1]: This PR work on PR https://github.com/dagger/dagger/pull/5565 which fixes this issue already. 🙇

**Comment by wingyplus** [comment]
@TomChv it's ready now. :)

**Comment by wingyplus** [comment]
I think rust sdk failure can fixes with https://github.com/dagger/dagger/pull/5626.

**Comment by wingyplus** [comment]
Do we need to add changie log here?

**Review by TomChv — APPROVED** [review]
LGTM, I see the Rust SDK tests are failing but it's not related to this PR.

Add a quick release note and we're good to merge

**Comment by wingyplus** [comment]
Sorry, i fallen into sleep last night. I think rebase on develop should be fix. I will investigate on it tonight.

**Comment by TomChv** [comment]
I see that Elixir SDK tests are failing, I ran these tests again to double check if it just a flacky behaviour or an actual issue

**Comment by wingyplus** [comment]
I cannot open my computer right now, I cannot get a full log on the CI. But I guess that it is because of connect timeout issue. The fix already applied since https://github.com/dagger/dagger/pull/5628. :)

**Comment by TomChv** [comment]
The Pr is already up to date with main, but the issue is still here :/

**Comment by wingyplus** [comment]
I'm wondering why this test spend so much time https://github.com/dagger/dagger/blob/main/sdk/elixir/test/dagger/client_test.exs#L165. 🤔

**Comment by wingyplus** [comment]
No it randomly failure. Open a bug https://github.com/dagger/dagger/issues/5642. Start investigating...

**Comment by TomChv** [comment]
I rebased the PR, I'm gonna merge that one if all checks are green 🚀

**Comment by wingyplus** [comment]
All green!!

**Pull request merged**
